core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedGlobalAggregationWithProjection.java

```diff
@@ -287,4 +298,15 @@ public Result apply(CorrelatedJoinNode correlatedJoinNode, Captures captures, Co
                 globalAggregation,
                 assignments));
     }
+
+    private static Optional<Symbol> getBooleanAggregationSymbol(AggregationNode node)
+    {
+        if (node.getAggregations().size() == 1) {
+            Aggregation aggregation = getOnlyElement(node.getAggregations().values());
+            if (aggregation.getResolvedFunction().name().equals(BOOL_OR) && aggregation.getArguments().getFirst() instanceof Reference symbolReference) {
```

Comment:
Please add tests to `TestCorrelatedAggregation`, where there is explicit `bool_or` in correlated subquery AND:
* subquery result is empty set, result should be `null` after decorrelation
* subquery input to `bool_or` is:
  * `null, false` -> result should be `false` after decorrelation
  * `false, true` -> result should be `true` after decorrelation
  * `null, true` -> result should be `true` after decorrelation
  * `null` -> result should be `null` after decorrelation